geoelements/gns
# Dev/grad checkpoint (#48)

* Rollout with gradient checkpointing

* Dev/enable cpu (#49)

* enable cpu training

* Update read me

* Remove grad checkpointing from this PR

---------



* Fix CPU CI testing (#50)

* Print test paths and sample folder

* Fix path to dataset in CI test

* Refactor training loop for CPU and GPU and cleanup only for GPU

* Add save files for cpu version (#51)

* Print test paths and sample folder

* Fix path to dataset in CI test

* Refactor training loop for CPU and GPU and cleanup only for GPU

* Test rollout prediction

* Try for EOF issue

* Try removing .git at clone step

* Fix rollout path

* Debug to see if model files are written

* Debug model path for rollout

* Save steps embedded in cpu mode as well

* Save model file in CPU

* Add instructions to test

* Dev/cpu env (#52)

* Rollout with gradient checkpointing

* enable cpu training

* Latest torch cluster installation in conda

* Remove unrelated files

* Add yes to installation with conda

* Use structured arrays to store positions and particle types (#53)

* Fixes #56 references.bib

* Fixes #57 paranthesis in references

* Fixes #54 DOIs

* Fixes #55 unclosed paranthesis in paper

* Update citation to v1.1.0

* Update title in citation

* Update citation file to point to doi on zeondo

* add a doc for explaining details about training data (#59)

* add a doc for explaining details about training data

* Add image

* Change heading levels

* Add MeshNet to _sidebar.md

* JOSS citation.cff file

* JOSS Citation in README

* Bug fix in cpu and gpu conditioning (#60)

* bug fix for cpu and gpu conditioning

* include loss in training_state file

* Use rank if cuda, else use device (which is cpu), to deal with different behaviors depending on cpu and gpu.

* reset flags

* JOSS citation.cff file

* JOSS Citation in README

* merge the recent upstream change

* rollback flags

* rollback flags

* typo

* bug fix for cpu and gpu conditioning

* include loss in training_state file

* Use rank if cuda, else use device (which is cpu), to deal with different behaviors depending on cpu and gpu.

* reset flags

* rollback flags

* rollback flags

* typo

* Define device id

---------



* Include grad checkpoint feature into the rollout function.

* `data_loader.py` accepts `.npz` with material property feature

* add boundary clamp limit option

* Enable `learned_simulator.py` to accept material property feature

* Add feature that can take optional metadata depending on whether `rollout` or `train` mode

* Update render_rollout.py

* Rollback to regular rollout without grad checkpoint, enable taking material property feature

* Rollback to regular `predict.py` without grad checkpoint, enable taking material property feature

* Enable `train.py` to take material property feature

* fixup! add boundary clamp limit option

* add another way of init simulator

* remove redundant parameters in `predict`

* add example for solving inverse problem using gns

* Remove previous grad checkpoint file

* Update requirements.txt

* Add inputfile flag

* Add config file for the inverse analysis

* Add readme

* Fix minor errors related to material property feature conditioning

* Update test

* Add figs for inverse example

* Update animation

* Update figs

* fix fig for initial config

* Move inverse example doc to `/docs`.

* Add rollout function's doc string

* Fix dataloader

---------

**Describe the PR**
A clear and concise description of the PR.

**Related Issues/PRs**
A clear and concise description of what you expected to happen.

**Additional context**
Add any other context about the problem here.
